src/lfortran/semantics/ast_common_visitor.h

```diff
@@ -3094,15 +3094,17 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                 if( m_value && m_value->type == ASR::exprType::StringConstant ) {
                     ASR::StringConstant_t *m_str = ASR::down_cast<ASR::StringConstant_t>(m_value);
                     std::string sliced_str;
+                    int str_length = strlen(m_str->m_s);
                     if( start <= 0 ) {
                         throw SemanticError("Substring `start` is less than one",
                                     loc);
                     }
+                    if(end > str_length) {
+                        throw SemanticError("Error: Substring end index exceeds the string length",
```

Comment:
Error can be removed from here
```suggestion
                        throw SemanticError("Substring end index exceeds the string length",
```

Reply:
Updated.